packages/cli/src/commands/deploy.ts

```diff
@@ -78,6 +80,24 @@ const printStartDeploy = async (output: CliOutput, executingDeploy: boolean, che
   }
 }
 
+const printDeploymentSummary = async (
+  summary: Record<DetailedChangeId, DeploySummaryResult>,
+  output: CliOutput,
+): Promise<void> => {
+  const summaryResultToId = Object.entries(summary).reduce(
+    (acc, [key, value]) => {
+      acc[value] = acc[value] || []
+      acc[value].push(key)
+      return acc
+    },
+    {} as Record<DeploySummaryResult, DetailedChangeId[]>,
```

Comment:
```suggestion
  const summaryResultToId: Record<DeploySummaryResult, DetailedChangeId[]> = Object.entries(summary).reduce(
    (acc, [key, value]) => {
      acc[value] = acc[value] || []
      acc[value].push(key)
      return acc
    },
```
you should avoid using casting here, maybe try adding the type in the declaration or inside the iterator

Reply:
I actually think that we can get this info from summarizeDeployChanges.
something like:
export const summarizeDeployChanges = (
  requested: Change[],
  applied: Change[],
): { a: Record<DetailedChangeId, DeploySummaryResult>; b: Record<DeploySummaryResult, DetailedChangeId[]> } => {
  const appliedById = _.keyBy(applied, c => getChangeData(c).elemID.getFullName())
  const a = Object.fromEntries(
    requested.flatMap(requestedChange => {
      const requestedChangeId = getChangeData(requestedChange).elemID.getFullName()
      return summarizeDeployChange(requestedChange, appliedById?.[requestedChangeId])
    }),
  )
  const b: Record<DeploySummaryResult, DetailedChangeId[]> = {
    success: [],
    failure: [],
    'partial-success': [],
  }

  Object.entries(a).forEach(([changeId, resultValue]) => {
    if (b[resultValue]) {
      b[resultValue].push(changeId)
    }
  })
  return { a, b }
}